### DISCUSSION TITLE
SDM72D-M V2 Modus problems with readings

### DISCUSSION THREAD

**Discussion by fireball79** [arendst/Tasmota#19178] [Support request]
Hi Guys,

i bought this energy meter in version 2, because i want to read the energy cosumption of my heatpump.
I use tasmota + IR together with SML to read my Holley counter and everything works fine. So i saw, SDM72D-M is also supported and i was thinking, cool, thats easy.

But now i invest many hours and i understand nothing anymore.

1. First try was to connect with this two adapter RS485 Adapters ([first ](https://www.ebay.de/itm/255283295160), [second](https://www.ebay.de/itm/255283194259)) aditionally to my IR-Reader the energy meter.
 I add only the reader pin to different Pins on my wemod d1 and i also used 3,3 and 5V.
Then i add the [SML Script ](https://tasmota.github.io/docs/Smart-Meter-Interface/#sdm72d-modbus), adjust the GPIO Port for ReaderPin, but nothing happend.

=> Questions: Can the SML Module Read the Modbus directly from the meter via RS485 adapter?
=> Questions How do i configure 2 Meters in SML? Is this correct:
`
>D
>B
->sensor53 r
>M 2
+1,14,s,16,9600,Haus
1,77070100600100ff@#,Zähler ID,,server_id,0
1,77070100020800ff@1000,Einspeisung (Total),kWh,export_total_kwh,4
1,77070100010800ff@1000,Verbrauch (Total),kWh,total_kwh,4
1,77070100100700ff@1,Verbrauch (Current),W,curr_w,0
1,77070100200700ff@1,Spannung L1,V,volt_p1,1
1,77070100340700ff@1,Spannung L2,V,volt_p2,1
1,77070100480700ff@1,Spannung L3,V,volt_p3,1
1,770701001f0700ff@1,Stromstärke L1,A,amp_p1,1
1,77070100330700ff@1,Stromstärke L2,A,amp_p2,1
1,77070100470700ff@1,Stromstärke L3,A,amp_p3,1
1,77070100510701ff@1,Phase angle U-L2/U-L1,deg,phase_angle_l2_l1,1
1,77070100510702ff@1,Phase angle U-L3/U-L1,deg,phase_angle_l3_l1,1
1,77070100510704ff@1,Phase angle I-L1/U-L1,deg,phase_angle_p1,1
1,7707010051070fff@1,Phase angle I-L2/U-L2,deg,phase_angle_p2,1
1,7707010051071aff@1,Phase angle I-L3/U-L3,deg,phase_angle_p3,1
1,770701000e0700ff@1,Frequenz,Hz,freq,0
+2,0,mN1,0,9600,WP,26,1,01040000,01040002,01040004,01040006,01040008,0104000a,0104000c,0104000e,01040010,01040012,01040014,01040016,01040018,0104001a,0104001c,0104001e,01040020,01040022,0104002a,0104002e,01040030,01040034,01040038,0104003c,0104003e,01040046,01040048,0104004A,01040156,01040158,0104018c,01040500,01040502
2,010404ffffffff@i0:1,Spannung P1,V,voltage_phase1,2
2,010404ffffffff@i1:1,Spannung P2,V,voltage_phase2,2
2,010404ffffffff@i2:1,Spannung P3,V,voltage_phase3,2
2,010404ffffffff@i3:1,Current P1,A,current_phase1,2
2,010404ffffffff@i4:1,Current P2,A,current_phase2,2
2,010404ffffffff@i5:1,Current P3,A,current_phase3,2
2,010404ffffffff@i6:1,Power P1,W,power_phase1,2
2,010404ffffffff@i7:1,Power P2,W,power_phase2,2
2,010404ffffffff@i8:1,Power P3,W,power_phase3,2
2,010404ffffffff@i9:1,Power Apparent P1,VA,power_apparent_phase1,2
2,010404ffffffff@i10:1,Power Apparent P2,VA,power_apparent_phase2,2
2,010404ffffffff@i11:1,Power Apparent P3,VA,power_apparent_phase3,2
2,010404ffffffff@i12:1,Power Reactive P1,VAr,power_reactive_phase1,2
2,010404ffffffff@i13:1,Power Reactive P2,VAr,power_reactive_phase2,2
2,010404ffffffff@i14:1,Power Reactive P3,VAr,power_reactive_phase3,2
2,010404ffffffff@i15:1,Power Factor P1,,power_factor_phase1,2
2,010404ffffffff@i16:1,Power Factor P2,,power_factor_phase2,2
2,010404ffffffff@i17:1,Power Factor P3,,power_factor_phase3,2
2,010404ffffffff@i18:1,Average Voltage,V,voltage_avg,2
2,010404ffffffff@i19:1,Average Current,A,current_avg,2
2,010404ffffffff@i20:1,Current Total,A,current_total,2
2,010404ffffffff@i21:1,Power Total,W,power_total,2
2,010404ffffffff@i22:1,Power Apparent Total,VA,power_a
#`

Then i activated this lines:
`
#define USE_SDM72                                // Add support for Eastron SDM72-Modbus energy monitor (+0k3 code)
#define SDM72_SPEED          9600              // SDM72-Modbus RS485 serial speed (default: 9600 baud)
#define SDM72_IMPEXP                            // Show additonal import/export active energy and power in MQTT and Web (+0k5 code)
`
I used reader and sender on GPIO 3 and 4 and setup in tasmota config the GPIOs for SDM Tx/Rx and now the was a comunication visible in the meter with telephon-symbol.
But also no data...

I do not found in the internet a configuration for reading 2 meters... one via IR and the other via Modbus.
Currently working is only the IR as single reader.

Can somebody help me to find the correct solution?!
Thanks a lot.

**Comment by mattaipm** [comment]
That's how it works. He worked with Domoticz for several months, but after one of the python updates there was a problem - the data stopped flowing.
